conda/base/context.py

```diff
@@ -850,17 +837,17 @@ def user_agent(self):
         builder.append("%s/%s" % self.os_distribution_name_version)
         if self.libc_family_version[0]:
             builder.append("%s/%s" % self.libc_family_version)
-        if self.solver.value != "classic":
+        if self.solver != "classic":
             from ..core.solve import _get_solver_class
 
-            user_agent_str = "solver/%s" % self.solver.value
+            user_agent_str = "solver/%s" % self.solver
             try:
                 # Solver.user_agent has to be a static or class method
                 user_agent_str += f" {_get_solver_class().user_agent()}"
             except Exception as exc:
                 log.debug(
                     "User agent could not be fetched from solver class '%s'.",
-                    self.solver.value,
+                    self.solver,
                     exc_info=exc
                 )
```

Comment:
If we pursue [this suggestion](https://github.com/conda/conda/issues/11918#issuecomment-1265619129), the whole user agent logic can be removed (a solver should not query anything on its own) because we would be using conda fetchers anyway.

Reply:
I don't anticipate any challenging changes. It might be just a single line, plus testing.

Reply:
That said, it might still be useful to keep user-agent info around to inform future choices like when/if to change the default solver choice. 

Alternatively, a user-agent plugin hook could be used? But perhaps this opens too many gates :D

Reply:
Yeah, a user-agent extension plugin hook has been discussed and we should make use of it going forward. But not in the coming release.